app/pages/sections/items/ItemsSection.scala

```diff
@@ -38,4 +38,23 @@ case object ItemsSection extends Section[JsObject] {
   }
 
   override def canBeCompletedForTraderAndDestinationType(implicit request: DataRequest[_]): Boolean = true
+
+  /**
+   * @param request user's DataRequest containing current UserAnswers
+   * @return for every item, for every packaging within that item,
+   *         return the ItemPackagingShippingMarksPage(itemIdx, packagingIdx) result if it exists
+   */
+  def retrieveAllShippingMarks()(implicit request: DataRequest[_]): Seq[String] =
+    request.userAnswers.get(ItemsCount)
+      .map {
+        itemsCount =>
+          (0 until itemsCount)
+            .flatMap(itemIdx => request.userAnswers.get(ItemsPackagingCount(itemIdx)).map {
+              packagingCount =>
+                (0 until packagingCount)
+                  .flatMap(packagingIdx => request.userAnswers.get(ItemPackagingShippingMarksPage(itemIdx, packagingIdx)))
+            })
+      }
+      .map(_.flatten.distinct)
+      .getOrElse(Seq())
```

Comment:
This has made me think about the following situations, what happens if:

- Item gets removed which was the one which initially had the shipping code?
- Item packaging gets removed which was the one which had the shipping code?
- Item packaging shipping code is then changed, after being selected in this drop down?

Would these scenarios invalidate the data if the user has entered if they picked one of the values which has either changed or no longer exists?

Reply:
a. Not sure; asked UCD
b. Not sure; asked UCD
c. The change should be reflected in this item/packaging, so my current solution of copying the value into a new variable doesn't work

Reply:
a. Will be dealt with in an alignment
b. Will be dealt with in an alignment
c. Fixed